pkg/helmreconciler/rendering.go

```diff
@@ -219,17 +225,19 @@ func (h *HelmReconciler) ProcessObject(chartName string, obj *unstructured.Unstr
 				}
 			}
 		}
-	} else if patch, err = h.CreatePatch(receiver, mutatedObj); err == nil && patch != nil {
-		log.Info("updating existing resource")
-		mutatedObj, err = patch.Apply()
-		if err == nil {
-			if err = h.customizer.Listener().ResourceUpdated(mutatedObj, receiver); err != nil {
-				log.Errorf("unexpected error occurred during postprocessing of updated resource: %s", err)
-			}
-		} else {
-			listenerErr := h.customizer.Listener().ResourceError(obj, err)
-			if listenerErr != nil {
-				log.Errorf("unexpected error occurred invoking ResourceError on listener: %s", listenerErr)
+	} else if update {
+		if patch, err = h.CreatePatch(receiver, mutatedObj); err == nil && patch != nil {
+			log.Info("updating existing resource")
+			mutatedObj, err = patch.Apply()
+			if err == nil {
+				if err = h.customizer.Listener().ResourceUpdated(mutatedObj, receiver); err != nil {
+					log.Errorf("unexpected error occurred during postprocessing of updated resource: %s", err)
+				}
+			} else {
+				listenerErr := h.customizer.Listener().ResourceError(obj, err)
+				if listenerErr != nil {
+					log.Errorf("unexpected error occurred invoking ResourceError on listener: %s", listenerErr)
```

Comment:
The same problem with error processing.  There are other instances of this poor error handling in this function that you didn't add.  Perhaps you can get it right in your changes and return with a followup PR to address the problems in the remainder of this code?

Cheers
-steve